app/src/main/java/tech/ula/utils/DownloadUtility.kt

```diff
@@ -57,18 +64,53 @@ class DownloadUtility(val context: Context, val archType: String, distType: Stri
         return downloadManager.enqueue(request)
     }
 
-    private fun assetNeedsToUpdated(type: String): Boolean {
+    private fun assetNeedsToUpdated(type: String, endpoint: String, newFilesystem: Boolean): Boolean {
+        val sharedPref = context.getSharedPreferences("file_date_stamps", Context.MODE_PRIVATE)
+
+        //TODO: make it so we download a full group of files, if any has changed (not the rootfs though, that is special)
+        //This will take care of a few possible corner cases
+
+        //only download the rootfs the first time, it cannot simply be updated like other files
+        if(type.contains("rootfs") && !newFilesystem) {
+            return false
+        }
+
+        //only download if this is a newFilesystem or it has been more than a day since we last checked
+        val now = java.util.Date().time
+        //TODO: should be store last updated in the session itself
+        val lastUpdated = sharedPref.getLong("lastUpdated", 0)
+        if (now < (lastUpdated + 86400000L) && !newFilesystem) {
+            return false
+        } else {
+            with (sharedPref.edit()) {
+                putLong("lastUpdated", now)
```

Comment:
If the download fails I don't think we'll check for a newer version if they retry it.

Reply:
that is true.  really need to update this later and just do the early exit here.